Source/NSTableCellView.m

```diff
@@ -0,0 +1,210 @@
+{
+  _rowSizeStyle = rowSizeStyle;
+}
+
+- (NSArray *) draggingImageComponents
+{
+  return _draggingImageComponents;
+}
+
+- (void) setDraggingImageComponents: (NSArray *)draggingImageComponents
+{
+  ASSIGNCOPY(_draggingImageComponents, draggingImageComponents);
+}
+
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+  [super encodeWithCoder: coder];
+  if ([coder allowsKeyedCoding])
+    {
+      [coder encodeObject: [self objectValue]
+		   forKey: @"NSObjectValue"];
+      [coder encodeObject: [self imageView]
+		   forKey: @"NSImageView"];
+      [coder encodeObject: [self textField]
+		   forKey: @"NSTextField"];
+      [coder encodeInt: [self backgroundStyle]
+		forKey: @"NSBackgroundStyle"];
+      [coder encodeInt: [self rowSizeStyle]
+		forKey: @"NSTableViewRowSizeStyle"];
+      [coder encodeObject: [self draggingImageComponents]
+		   forKey: @"NSDraggingImageComponents"];
+    }
+  else
+    {
+      [coder encodeObject: _objectValue];
+      [coder encodeObject: _imageView];
+      [coder encodeObject: _textField];
+      [coder encodeObject: _draggingImageComponents];
+
+      [coder encodeValueOfObjCType: @encode(NSBackgroundStyle)
+				at: &_backgroundStyle];
+      [coder encodeValueOfObjCType: @encode(NSTableViewRowSizeStyle)
+				at: &_rowSizeStyle];
+    }
+}
+
+- (id) initWithCoder: (NSCoder *)coder
+{
+  self = [super initWithCoder: coder];
+  if (self != nil)
+    {
+      if ([coder allowsKeyedCoding])
+	{
+	  if ([coder containsValueForKey: @"NSObjectValue"])
+	    {
+	      [self setObjectValue: [coder decodeObjectForKey: @"NSObjectValue"]];	      
+	    }
+
+	  if ([coder containsValueForKey: @"NSImageView"])
+	    {
+	      [self setImageView: [coder decodeObjectForKey: @"NSImageView"]];	      
+	    }
+
+	  if ([coder containsValueForKey: @"NSTextField"])
+	    {
+	      [self setTextField: [coder decodeObjectForKey: @"NSTextField"]];	      
+	    }
+	  
+	  if ([coder containsValueForKey: @"NSBackgroundStyle"])
+	    {
+	      [self setBackgroundStyle: [coder decodeIntForKey: @"NSBackgroundStyle"]];
+	    }
+
+	  if ([coder containsValueForKey: @"NSTableViewRowSizeStyle"])
+	    {
+	      [self setRowSizeStyle: [coder decodeIntForKey: @"NSTableViewRowSizeStyle"]];
+	    }
+
+	  if ([coder containsValueForKey: @"NSDraggingImageComponents"])
+	    {
+	      [self setDraggingImageComponents: [coder decodeObjectForKey: @"NSDraggingImageComponents"]];
+	    }
+	}
+      else
+	{
+	  [self setObjectValue: [coder decodeObject]];
+	  [self setImageView: [coder decodeObject]];
+	  [self setTextField: [coder decodeObject]];
+	  [self setDraggingImageComponents: [coder decodeObject]];
+	  
+	  [coder decodeValueOfObjCType: @encode(NSBackgroundStyle)
+				    at: &_backgroundStyle];
+	  [coder decodeValueOfObjCType: @encode(NSTableViewRowSizeStyle)
+				    at: &_rowSizeStyle];	  
+	}
+    }
+  return self;
+}
+
+- (id) copyWithZone: (NSZone *)zone
+{
+  NSData *d = [NSArchiver archivedDataWithRootObject: self];
```

Comment:
This is the version of copy that we normally try to avoid. Why is it needed here?

Reply:
I will change it.

Reply:
Done